interface.go

```diff
@@ -95,7 +79,7 @@ func (c *Cache[T]) GetOrCompute(key string, evaluator func() (*T, error)) (*T, e
 		// Key not found on cache
 		go func() {
 			// Set key to cache in gorutine
-			c.Set(key, calculatedValue)
+			c.setNoLock(key, calculatedValue)
```

Comment:
I don't think this will work, as you acquire the lock in a different go routine and it could long be released at this point